### PR TITLE
ROC-3671: Move front-end validators in common repo

### PR DESCRIPTION
**Pull request by ellie-harrison**
PR to remove common validators from project and replace them with validators found in the `@hmcts/cmc-validators` npm package.

### CODE REVIEW

File: src/main/app/forms/models/contactDetails.ts [hunk 1]
```diff
@@ -12,10 +11,11 @@ export class ValidationErrors {
 export class ContactDetails implements Serializable<ContactDetails> {
 
   @IsOptional()
-  @IsPhone({ message: ValidationErrors.PHONE_NUMBER_NOT_VALID })
+  @IsPhoneNumber({ message: ValidationErrors.PHONE_NUMBER_NOT_VALID })
   phoneNumber?: string
 
   @IsOptional()
+  @IsNotEmpty()
```
Comment (geek-soft) [inline src/main/app/forms/models/contactDetails.ts]: I bet we can remove it. Empty value is deserialised to undefined which is permitted. Value with single whitespaces only is not forbidden by email validator.

File: src/test/app/forms/models/amount.ts [hunk 2]
```diff
@@ -114,12 +114,12 @@ describe('Amount', () => {
     it('should reject Nan Upper value', () => {
```
Comment (geek-soft) [inline src/test/app/forms/models/amount.ts]: I recon we should remove that test case as it is impossible to have NaN in current implementation

File: src/main/app/forms/models/contactDetails.ts (hunk 1 above)
Comment (ellie-harrison) [reply src/main/app/forms/models/contactDetails.ts]: There is a unit test that checks that an empty email address is rejected (Line 80 here -> https://github.com/hmcts/cmc-legal-rep-frontend/blob/master/src/test/app/forms/models/contactDetails.ts). 

Should this be left as is, or have the unit test removed if it is irrellevant?

Comment (geek-soft) [reply src/main/app/forms/models/contactDetails.ts]: That tests did not reflect real use so I removed it. The way deserialisation is implemented makes empty email addresses to be resolved to undefined and the validation rules should permit such case.

**Pull request merged**